gazebo_plugins/src/gazebo_ros_camera.cpp

```diff
@@ -407,13 +403,21 @@ void GazeboRosCamera::Load(gazebo::sensors::SensorPtr _sensor, sdf::ElementPtr _
     if (impl_->camera_[i]->LensDistortion()) {
       impl_->camera_[i]->LensDistortion()->SetCrop(border_crop);
 
+      // Get Optical center from camera
+      default_cx = impl_->camera_[i]->LensDistortion()->Center().X();
+      default_cy = impl_->camera_[i]->LensDistortion()->Center().Y();
```

Comment:
Based on my unit testing it looks like the way we are getting the` cx, cy` from lens distortion is also wrong. I guess the `center `tag of `distortion` tag takes values from **0 to 1** and we need to multiple by corresponding height and width to get `cx, cy.`

Also, if no `center` tag is provided in distortion tag it defaults to **0.5** for both cx and cy.

Reply:
cc @adityapande-1995 @jacobperron

Reply:
Is there another SDF element that we can check? [This ticket](https://github.com/gazebosim/sdformat/issues/1071) implies that there is a <camera><lens><intrinsics><cx> is also valid. What do the intrinsic camera center values default to?

Reply:
Based on my testing it seems like there isn't any other SDF element which we can use. There is a **ProjectionMatrix()** in gazebo which we can probably use to get some information. However, the **ProjectionMatrix** is calculated in a weird normalised way from OGRE. In case when no `<camera><lens><intrinsics><cx>` is provided the gazebo calculates the projection matrix using other parameters(image size, focal length and fov). 

Here is where calculation for projection matrix is done in case **when no intrinsic are provided.**
https://github.com/OGRECave/ogre/blob/36c8fe6ae25fc36d039dc542669809e365cb21fb/OgreMain/src/OgreFrustum.cpp#L325-L336